streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -820,110 +778,6 @@ public synchronized <KIn, VIn> Topology addReadOnlyStateStore(final StoreBuilder
-     * records forwarded from the {@link SourceNode}.
-     * The supplier should always generate a new instance each time
-     * {@link org.apache.kafka.streams.processor.ProcessorSupplier#get()} gets called. Creating a single
-     * {@link org.apache.kafka.streams.processor.Processor} object and returning the same object reference in
-     * {@link org.apache.kafka.streams.processor.ProcessorSupplier#get()} would be a violation of the supplier pattern
-     * and leads to runtime exceptions.
-     * This {@link ProcessorNode} should be used to keep the {@link StateStore} up-to-date.
-     *
-     * @param storeBuilder          user defined key value store builder
-     * @param sourceName            name of the {@link SourceNode} that will be automatically added
-     * @param timestampExtractor    the stateless timestamp extractor used for this source,
-     *                              if not specified the default extractor defined in the configs will be used
-     * @param keyDeserializer       the {@link Deserializer} to deserialize keys with
-     * @param valueDeserializer     the {@link Deserializer} to deserialize values with
-     * @param topic                 the topic to source the data from
-     * @param processorName         the name of the {@link org.apache.kafka.streams.processor.ProcessorSupplier}
-     * @param stateUpdateSupplier   the instance of {@link org.apache.kafka.streams.processor.ProcessorSupplier}
-     * @return itself
-     * @throws TopologyException if the processor of state is already registered
-     * @deprecated Since 2.7.0. Use {@link #addGlobalStore(StoreBuilder, String, TimestampExtractor, Deserializer, Deserializer, String, String, ProcessorSupplier)} instead.
-     */
-    @Deprecated
-    public synchronized <K, V> Topology addGlobalStore(final StoreBuilder<?> storeBuilder,
-                                                       final String sourceName,
-                                                       final TimestampExtractor timestampExtractor,
-                                                       final Deserializer<K> keyDeserializer,
-                                                       final Deserializer<V> valueDeserializer,
-                                                       final String topic,
-                                                       final String processorName,
-                                                       final org.apache.kafka.streams.processor.ProcessorSupplier<K, V> stateUpdateSupplier) {
-        internalTopologyBuilder.addGlobalStore(
-            new StoreBuilderWrapper(storeBuilder),
-            sourceName,
-            timestampExtractor,
-            keyDeserializer,
-            valueDeserializer,
-            topic,
-            processorName,
-            () -> ProcessorAdapter.adapt(stateUpdateSupplier.get()),
```

Comment:
Do we use `ProcessorAdapter.adapt(...)` elsewhere? If not, we can remove it, too. (Maybe if a follow up PR if this PR becomes too big -- it's simple, but it's already more than 700 LOC...)